lightningd/pay.c

```diff
@@ -42,6 +42,31 @@ struct sendpay_command {
 	struct command *cmd;
 };
 
+static enum wallet_payment_status string_to_payment_status(const char *status_str)
```

Comment:
The usual convention for parsers that can fail is something like this:

```c
bool string_to_payment_status(const char *status_str, enum wallet_payment_status *)
```

Then later in use:

```
if (!string_to_payment_status(status_str, &status))
    return command_fail(cmd, JSONRPC2_INVALID_PARAMS, "Unrecognized status '%s'", status_str);
```

Reply:
Changed https://github.com/ElementsProject/lightning/commit/627a374820dd03876eae9e4a99d661cb39f63934